### PR TITLE
Hide category

### PR DESCRIPTION
**Pull request by alschmiedt**
## The basics

- [ ] I branched from develop
- [ ] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
Adds ability to hide and disable a category. 

### Proposed Changes
Add isHidden_ and isDisabled properties.

### Test Coverage

Tested on:

### Additional Information
If isHidden is set to true on the currently selected item, this relies on the developer to call toolbox.clearSelection() to close the flyout.

### CODE REVIEW

File: core/toolbox/category.js [hunk 1]
```diff
@@ -126,11 +126,27 @@ Blockly.ToolboxCategory = function(categoryDef, toolbox, opt_parent) {
...
+   * True if the parent category is expanded, false otherwise.
+   * Children categories can only be visible if their parent category is
+   * expanded.
+   * @type {boolean}
+   * @private
+   */
+  this.isParentExpanded_ = true;
+
+  /**
+   * True if this category is not disabled, false otherwise.
```
Comment (maribethb) [inline core/toolbox/category.js:145]: -not

File: core/toolbox/category.js [hunk 2]
```diff
@@ -494,24 +510,44 @@ Blockly.ToolboxCategory.prototype.hasChildren = function() {
...
+  if (this.hasChildren()) {
+    for (var i = 0, child; (child = this.contents_[i]); i++) {
+      child.setVisible_(isVisible);
+    }
+  }
+  this.isHidden_ = !isVisible;
+};
+
+/**
+ * Hide the category.
```
Comment (maribethb) [inline core/toolbox/category.js:529]: Question: In the PR description you said in certain cases the developer would need to call toolbox.clearSelection() to close the flyout; is that the case when they call this hide function on the currently selected item? If so, maybe add that instruction here so they know they need to call it?

Or is it possible to detect that this is being called on the selected item and call it for them? I wasn't sure if that was a requirement for them to call the function or if you were clarifying that if they wanted the flyout to close, they would still have to do it themselves.

Comment (alschmiedt) [reply core/toolbox/category.js:529]: So, it doesn't technically break anything but it is definitely a bit weird to have the flyout open when the category has been hidden. 

We could do something like: 
```
if (this.parentToolbox_.getSelectedItem() == this) {
   this.parentToolbox_.clearSelection();
}
```
It seemed a bit messy and wasn't sure if we wanted to leave this up to the developer. But thinking about it more I can't think of a situation where a developer wouldn't want to do this. So I think I will add back in.

File: core/toolbox/category.js (hunk 1 above)
Comment (alschmiedt) [reply core/toolbox/category.js]: done

**Pull request merged**